third-party/realdds/src/dds-stream-server.cpp

```diff
@@ -202,24 +201,66 @@ void dds_stream_server::close()
     _writer.reset();
 }
 
-void dds_stream_server::publish( const uint8_t * data, size_t size, unsigned long long id )
+void dds_video_stream_server::publish_image( topics::image_msg && image )
 {
     if( ! is_streaming() )
         DDS_THROW( runtime_error, "stream '" + name() + "' cannot publish before start_streaming()" );
 
+    if( image.height != _image_header.height )
+        DDS_THROW( runtime_error,
+                   "image height (" + std::to_string( image.height ) + ") does not match stream header ("
+                       + std::to_string( _image_header.height ) + ")" );
+    if( image.width != _image_header.width )
+        DDS_THROW( runtime_error,
+                   "image width (" + std::to_string( image.width ) + ") does not match stream header ("
+                       + std::to_string( _image_header.width ) + ")" );
+
+    // LOG_DEBUG( "publishing a DDS video frame for topic: " << _writer->topic()->get()->get_name() );
+    sensor_msgs::msg::Image raw_image;
+    raw_image.header().frame_id() = image.frame_id;
+    raw_image.header().stamp().sec() = image.timestamp.seconds;
+    raw_image.header().stamp().nanosec() = image.timestamp.nanosec;
+    raw_image.encoding() = _image_header.format.to_string();
+    raw_image.height() = _image_header.height;
+    raw_image.width() = _image_header.width;
+    raw_image.step() = uint32_t( image.raw_data.size() / _image_header.height );
+    raw_image.is_bigendian() = false;
```

Comment:
Is true for Jetson?

Reply:
It's reflective of our images (whoever created the data) rather than the system. For now it should do.

Reply:
Actually, I'm not entirely sure what it's indicative of. It might refer to the endian-ness of the message header, since the image format is pretty much set in stone...
I think we'll address this issue when we get there.